bob/src/api/mod.rs

```diff
@@ -313,31 +319,69 @@ async fn status<A: Authenticator>(bob: Extension<BobServer<A>>) -> Json<Node> {
 async fn get_space_info<A: Authenticator>(
     bob: Extension<BobServer<A>>,
 ) -> Result<Json<SpaceInfo>, StatusExt> {
+    let mut disk_metrics = bob.grinder().hw_counter().update_space_metrics();
     let DiskSpaceMetrics {
         total_space,
         used_space,
         free_space,
-    } = bob.grinder().hw_counter().update_space_metrics();
+    } = disk_metrics.values().sum();
 
     let backend = bob.grinder().backend().inner();
     let (dcs, adc) = backend
         .disk_controllers()
         .ok_or_else(not_acceptable_backend)?;
-    let mut map = HashMap::new();
-    for dc in dcs.iter() {
-        map.insert(dc.disk().name().to_string(), dc.disk_used().await);
+    let mut disk_space_by_disk = HashMap::new();
+    let mut total_occupied = 0;
+    for dc in dcs {
+        let occupied_space = dc.disk_used().await;
+        total_occupied += occupied_space;
+        let DiskSpaceMetrics {
+            total_space,
+            used_space,
+            free_space,
```

Comment:
This decomposition is not needed, because below you create a new struct anyway. That new struct can be easily initialized from the fields of `DiskSpaceMetrics`